packages/aws-cdk/e2e-tests/package.json

```diff
@@ -0,0 +1,35 @@
+{
+  "name": "e2e-tests",
+  "description": "The end-to-end tests for the AWS CDK (libraries and CLI)",
+  "version": "0.0.0",
+  "private": true,
+  "scripts": {
+    "build": "tsc",
+    "watch": "tsc -w",
+    "test": "jest --config jest.config.unit.js",
+    "build+test": "npm run build && npm run test",
+
+    "test-e2e-against-repo": "./run-wrappers/run-against-repo jest --runInBand --config jest.config.e2e.js",
+    "test-e2e-against-dist": "./run-wrappers/run-against-dist jest --runInBand --config jest.config.e2e.js",
+    "test-e2e-against-release": "./run-wrappers/run-against-release jest --runInBand --config jest.config.e2e.js",
+
+    "test-init-against-dist": "./run-wrappers/run-against-dist test-init/test-all.sh",
+
+    "integ-cli": "npm run integ-cli-regression && npm run test-e2e-against-repo",
+    "integ-cli-regression": "npm run integ-cli-regression-latest-release && npm run integ-cli-regression-latest-code",
+    "integ-cli-regression-latest-release": "./run-wrappers/run-against-dist test-cli-regression-against-latest-release.sh",
+    "integ-cli-regression-latest-code": "./run-wrappers/run-against-dist test-cli-regression-against-current-code.sh"
+  },
+  "dependencies": {
+    "aws-sdk": "^2.804.0",
+    "jest": "^26.6.3",
+    "jest-junit": "^12.0.0",
+    "serve-npm-tarballs": "^0.1.12"
+  },
+  "devDependencies": {
+    "@types/jest": "^26.0.15",
```

Comment:
I wonder if this doesn't need a dependency on `aws-cdk` ? How else would you guarantee that `aws-cdk` gets built before the tests are run?

Reply:
I actually purposefully omitted this dependency, because I thought it would interfere with the "Unzip & run `yarn install && yarn test-e2e-against-dist`" experience that I wanted  in our canaries and pipeline jobs (it would report a "Could not find package 'aws-cdk@0.0.0'" error). Your concern:

> How else would you guarantee that `aws-cdk` gets built before the tests are run?

is only really a problem for local development (`run-against-repo`), as this package doesn't depend on any code in `aws-cdk` (neither in `lib`, nor in `test`), and during local development when using `run-against-repo` I assume people would run `yarn build / watch` in `aws-cdk` constantly anyway.

--------------------------

Having said all of that 🙂 I kind of already divided the dependencies into `"dependencies"` and `"devDependencies"`, with the thought of changing the canaries and pipeline jobs to use `yarn install --production=true` (to avoid downloading any unnecessary dependencies). So, I guess we could add `aws-cdk@0.0.0` to the `"devDependencies"` section here, and it wouldn't cause a problem.... how does that sound?

Reply:
> I assume people would run yarn build / watch in aws-cdk constantly anyway.

No assumptions on workflow please. First of all, it needs to work from a clean build, and second of all all these implicit ordering assumptions and non-declared dependencies fight with our a way to do clean, well-specified, hermetic builds (that can then subsequently be cached etc).

Honestly, it seems correct for this package to have a dependency on `aws-cdk`, because it runs it. If we did that, it would mean that we could even get rid of the `npm install aws-cdk` commands in there (although I guess potentially not because we also have regression tests which need their own fine-grained control over the CLI version).

> it would report a "Could not find package 'aws-cdk@0.0.0'" error

Would it? The `0.0.0` should have been replaced at release time, so it shouldn't really.

> So, I guess we could add aws-cdk@0.0.0 to the "devDependencies" section here

Yes, let's do at least that.